os/kernel/debug/Make.defs

```diff
@@ -15,6 +15,9 @@
 # language governing permissions and limitations under the License.
 #
 ###########################################################################
+ifeq ($(CONFIG_DEBUG)$(CONFIG_CPP_HAVE_VARARGS),yn)
+CSRCS += debug.c
+endif
```

Comment:
The debug.c file uses `CONFIG_SECURE_DEBUG_ASSERT` internally. If the config is disable, the .c file is nothing. But Make.defs uses different configs to include the .c file. You need to use the same conditional. If you keep like this, user even enables `CONFIG_SECURE_DEBUG_ASSERT`, this will not work.

Reply:
Thank you. I understand, changed.